lib/puppet/pops/parser/egrammar.ra

```diff
@@ -655,7 +655,8 @@ classname
   : NAME
   | WORD
   | CLASSREF
-  | CLASS { error val[0], "'class' is not a valid classname" }
+  | CLASS { error val[0], "'class' keyword not allowed at this location" }
+  | STRING { error val[0], "a string was found where it is not accepted" }
```

Comment:
you forgot to update the tests. Also, my second suggestion is better as it includes the information that the error is about a "name"